### PR TITLE
added new array board type

### PR DESCRIPTION
**Pull request by andrewjmeier**
@konnorbeard

### CODE REVIEW

File: src/checkers/checkers_board_dup.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  this.buildTokenForSpace(space, TableTop.Constants.redColor);
+
+  space = this.getSpace(22);
+  this.buildTokenForSpace(space, TableTop.Constants.whiteColor);
+};
+
+// creates the token for given space and color, 
+// adds it to the space, 
+// and appends it to our list of tokens
+CheckerBoard.prototype.buildTokenForSpace = function(space, color) {
```
Comment (konnorbeard) [inline src/checkers/checkers_board_dup.js]: Should we abstract this somewhere? feel like i've seen it before

File: tabletop/core/array_board.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+inherits(ArrayBoard, Board);
+
+ArrayBoard.prototype.getSpace = function(index) { 
+  return this.spaces[index]
+};
+
+ArrayBoard.prototype.getSpacePosition = function(space) { 
+
+  for (var i = 0; i < this.spaces.length; i++) {
```
Comment (konnorbeard) [inline tabletop/core/array_board.js]: No change needed here, but you should check out for each loops. can do cool stuff ie. 

    this.spaces.forEach(function(space) { 
        if space == match....
    )} 

just looks nicer/more pro imo

File: tabletop/core/view.js [hunk 3]
```diff
@@ -24,11 +25,11 @@ function View(game, turnMap) {
 View.prototype.drawBoard = function() { 
   if (this.game.board instanceof GridBoard) 
     this.drawGridBoard();
-  
+  else if (this.game.board instanceof ArrayBoard) {
```
Comment (konnorbeard) [inline tabletop/core/view.js]: nice

File: tabletop/core/view.js [hunk 4]
```diff
@@ -50,8 +51,65 @@ View.prototype.drawGridBoard = function() {
...
+
+};
+
+View.prototype.drawArrayTiles = function() {
+  var tileWidth = c.boardWidth / this.game.board.width;
+  var tileHeight = c.boardHeight / this.game.board.height;
+  
+  var tileNum = 0;
+
+  for (var x = 0; x < this.game.board.width; x++) {
```
Comment (konnorbeard) [inline tabletop/core/view.js]: Is there anyway we could abstract some of this? seems like alot of repeated functionaliy. maybe these lines: 
 +     var tileView = this.drawTile(tile, {width: tileWidth, height: tileHeight});
 +    tileView.x = (x * tileWidth);
 +    tileView.y = 0;
 +    this.tileViews[tileNum] = tileView;
 +    this.boardView.addChild(tileView);

**Comment by konnorbeard** [comment]
@andrewjmeier lgtm see above

File: src/checkers/checkers_board_dup.js (hunk 1 above)
Comment (andrewjmeier) [inline src/checkers/checkers_board_dup.js]: Already done https://github.com/andrewjmeier/TableTop.JS/blob/master/tabletop/core/board.js but checkers didn't get updated I think. I'm not sure if I even like it as a method of the board though. Might be better as a method on token or space

**Pull request merged**